### DISCUSSION TITLE
Validation before topics subscription

### DISCUSSION THREAD

**Discussion by YassinEldeeb** [MichalLytek/type-graphql#946] [Q&A] [answered]
now I want to make subscription for post comments, so I take a postId and publish every comment is written on it to the subscribed clients, the problem is I want to do some validation to ensure that the post the user want to subscribe to its comments exists.

So I read all of the docs on TypeGraphQL Subscriptions, and wrote my first subscription and it works so well, but the problem I wanna throw an error to the client if the post doesn't exist for the provided postId,

1- I tried in the topics function to find the post if it exists return the topic if not throw an error but the problem is it crashes the whole server and don't send any error to the requested client.

2- then I already have a `@PostExists()` custom decorator so I to made a PostIdInput `ArgsType`  and stick that `@PostExists()` on top of the id `field` and used that `ArgsType` in my comment resolver but the problem is that the comment resolver down below only runs when new data is published and not when subscribing, the `@Subscription` decorator runs when I subscribe to a topic, so any Validation I would stick it to the comment resolver arguments will not affect the subscription above.

```
class CommentsSubscriptionsResolver {
  @Subscription((_returns) => CommentSubscriptionPayload, {
    topics: ({ args }) => `${Topics.CommentsOnPost}:${args.postId}`,
  })
  comment(
    @Root() data: CommentSubscriptionPayload,
    @Arg('postId') _id: string
  ): CommentSubscriptionPayload {
    return data
  }
}
```

so how can I run a validation to make sure that post exists before subscribing, and throw an error to the client if the post doesn't exist.

**Comment by MichalLytek** [comment, marked as answer]
Use `subscribe` option of `@Subscription` decorator to perform custom operations and actions during the subscribe phase.

Note that you need to call `pubSub` with filters and topics manually.

**Reply by YassinEldeeb** [reply]
Thanks, it works.
Sorry for making it as an issue and not discussion, I didn't know.
